#COVID-19 Dataset  Analysis, Visualization, Prediction & Comparisons

Data Source (Date wise) : 2019 Novel Coronavirus COVID-19 (2019-nCoV) Data Repository by Johns Hopkins CSSE

Data Source: https://github.com/CSSEGISandData/COVID-19/tree/master/csse_covid_19_data/csse_covid_19_daily_reports

In [0]:
from IPython.core.interactiveshell import InteractiveShell
InteractiveShell.ast_node_interactivity = "all"

import warnings
warnings.filterwarnings('ignore')

import pandas as pd

pd.options.display.max_columns
pd.set_option('display.max_columns', None)
pd.options.display.max_rows
pd.set_option('display.max_rows', 5)

data = pd.read_csv('https://raw.githubusercontent.com/CSSEGISandData/COVID-19/master/csse_covid_19_data/csse_covid_19_daily_reports/03-17-2020.csv')

**Task 1.** Write a Python program to display first 5 rows from COVID-19 dataset. Also
print the dataset information and check the missing values.

In [0]:
print('\nShape of Data: ', data.shape[0], 'rows &', data.shape[1], 'columns')

print('\nData Information: ')
data.info()

print('\nTotal number of NaNs in Data (column-wise):\n')
data.isna().sum()


Shape of Data:  276 rows & 8 columns

Data Information: 
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 276 entries, 0 to 275
Data columns (total 8 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Province/State  126 non-null    object 
 1   Country/Region  276 non-null    object 
 2   Last Update     276 non-null    object 
 3   Confirmed       276 non-null    int64  
 4   Deaths          276 non-null    int64  
 5   Recovered       276 non-null    int64  
 6   Latitude        276 non-null    float64
 7   Longitude       276 non-null    float64
dtypes: float64(2), int64(3), object(3)
memory usage: 17.4+ KB

Total number of NaNs in Data (column-wise):



Province/State    150
Country/Region      0
                 ... 
Latitude            0
Longitude           0
Length: 8, dtype: int64

**Task 2:** Write a Python program to get the number of occurences of each country & the latest number of confirmed, deaths, recovered and active cases of Novel Coronavirus (COVID-19) country wise.

In [0]:
print('Number of occurences of each country:')
data['Country/Region'].value_counts().sort_index()

# Creation of "Active" column
data['Active'] = data['Confirmed'] - (data['Deaths'] + data['Recovered'])

print('\n\nLatest Trends (Confirmed, Deaths, Recovered, Active Cases) Country Wise:')
data.groupby('Country/Region')['Confirmed', 'Deaths', 'Recovered', 'Active'].sum()

Number of occurences of each country:


Afghanistan                       1
Albania                           1
                                 ..
Vietnam                           1
occupied Palestinian territory    1
Name: Country/Region, Length: 159, dtype: int64



Latest Trends (Confirmed, Deaths, Recovered, Active Cases) Country Wise:


Confirmed  Deaths  Recovered  Active
Country/Region                                                      
Afghanistan                            22       0          1      21
Albania                                55       1          0      54
...                                   ...     ...        ...     ...
Vietnam                                66       0         16      50
occupied Palestinian territory          0       0          0       0

[159 rows x 4 columns]

**Task 3:** Write a Python program to get the latest number of active cases, confirmed cases, deaths and recovered people of Novel Coronavirus (COVID-19) cases Country/Region - Province/State wise.

In [0]:
print('Latest Trends (Active, Confirmed, Deaths, Recovered Cases) Country & Region Wise:')
data.groupby(['Country/Region', 'Province/State'])['Active', 'Confirmed', 'Deaths', 'Recovered'].max()

Latest Trends (Active, Confirmed, Deaths, Recovered Cases) Country & Region Wise:


Active  Confirmed  Deaths  \
Country/Region Province/State                                            
Australia      Australian Capital Territory       2          2       0   
               From Diamond Princess              0          0       0   
...                                             ...        ...     ...   
United Kingdom Gibraltar                          2          3       0   
               United Kingdom                  1843       1950      55   

                                             Recovered  
Country/Region Province/State                           
Australia      Australian Capital Territory          0  
               From Diamond Princess                 0  
...                                                ...  
United Kingdom Gibraltar                             1  
               United Kingdom                       52  

[126 rows x 4 columns]

**Task 4:** Write a Python program to get the Chinese province wise cases of confirmed, deaths and recovered cases of Novel Coronavirus (COVID-19).

In [0]:
print('Latest Trends (Active, Confirmed, Deaths, Recovered Cases) in China (Region Wise):')
china = data[data['Country/Region'] == 'China']
china = china[['Province/State', 'Active', 'Confirmed', 'Deaths', 'Recovered']]
china.sort_values(by='Province/State').reset_index(drop=True)

Latest Trends (Active, Confirmed, Deaths, Recovered Cases) in China (Region Wise):


Province/State  Active  Confirmed  Deaths  Recovered
0           Anhui       0        990       6        984
1         Beijing      79        456       8        369
..            ...     ...        ...     ...        ...
31         Yunnan       2        176       2        172
32       Zhejiang      15       1232       1       1216

[33 rows x 5 columns]

**Task 5:** Write a Python program to get the latest country wise deaths cases of Novel Coronavirus (COVID-19).

**Task 6:** Write a Python program to list countries with no cases of Novel Coronavirus (COVID-19) recovered.

**Task 7:** Write a Python program to list countries with all cases of Novel Coronavirus (COVID-19) died.

**Task 8:** Write a Python program to list countries with all cases of Novel Coronavirus (COVID-19) recovered.

**Task 9:** Write a Python program to get the top 10 countries data (Last Update, Country/Region, Confirmed, Deaths, Recovered) of Novel Coronavirus (COVID-19).

**Task 10:** Write a Python program to create a plot (lines) of total deaths, confirmed, recovered and active cases Country wise where deaths greater than 150.

**Task 11:** Write a Python program to visualize the state/province wise death cases of Novel Coronavirus (COVID-19) in USA.

**Task 12:** Write a Python program to visualize the state/province wise Active cases of Novel Coronavirus (COVID-19) in USA.

**Task 13:** Write a Python program to visualize the state/province wise combine number of confirmed, deaths, recovered, active Novel Coronavirus (COVID-19) cases in USA.

**Task 14:** Write a Python program to visualize Worldwide Confirmed Novel Coronavirus (COVID-19) cases over time.